language/stdlib/modules/libra_configs.move

```diff
@@ -31,38 +37,58 @@ module LibraConfig {
         });
     }
 
-    // We can't return reference here, this returns a copy of the config value.
-    public fun get<Config: copyable>(addr: address): Config acquires T {
+    public fun apply_for_default_privilege() {
+        Association::apply_for_association();
+        Association::apply_for_privilege<ConfigCreatorPrivilege>();
+    }
+
+    public fun grant_default_privilege(addr: address) {
+        Association::grant_association_address(addr);
+        Association::grant_privilege<ConfigCreatorPrivilege>(addr);
+    }
+
+    // Get a copy of `Config` value stored under `addr`.
+    public fun get<Config: copyable>(): Config acquires T {
+        let addr = default_config_address();
         Transaction::assert(::exists<T<Config>>(addr), 24);
         *&borrow_global<T<Config>>(addr).payload
     }
 
     // Set a config item to a new value and trigger a reconfiguration.
-    public fun set<Config: copyable>(addr: address, payload: Config) acquires T, Configuration {
-        // TODO: impose proper permission checks
-        // Callable by the any address for now.
-
+    public fun set<Config: copyable>(_cap: &ConfigModifierPrivilege<Config>, payload: Config) acquires T, Configuration {
```

Comment:
If we changed this API to `public fun set<Config: copyable>(payload: Config)` and checked `exists<ConfigModifierCapability<Config>>(sender)` in the body, could we get rid of this unfortunate `CapabilityHolder` pattern?

Even if we need that pattern in some places (I suspect we do for validator key rotation), if we could provide two APIs for `set` (one that checks `exist` and one that takes the capability as a parameter) that gets rid of most uses of the pattern, I think that would be a nice simplification.

Reply:
You are indeed right about this. This pattern will fail when we try to rotate validator key. I do like the two api suggestion. I would propose the following api:
Either:
``` 
fun publish_new_config<Config>(t: Config): ModifyConfigCapability<Config>;
fun store_config_capability<Config>(cap: ModifyConfigCapability<Config>);
fun set<Config>(t: Config);
fun set_with_capability<Config>(cap: & ModifyConfigCapability<Config>, t: Config);
```
Or:
```
fun publish_new_config_with_default_capability<Config>(t: Config);
fun publish_new_config<Config>(t: Config): ModifyConfigCapability<Config>;
fun set<Config>(t: Config);
fun set_with_capability<Config>(cap: & ModifyConfigCapability<Config>, t: Config);
```

The problem to both approaches is that we loose the ability to delegate the address in the default mode. To modify configs published in such way, the transaction sender has to be config address (0xF1A95). We might want to have a default delegator so that one other address will be able to modify such value as well. So it will look something like:
 ```
fun publish_new_config_with_default_capability<Config>(t: Config, owner_address: Address);
// Only owner_address can call this function 
fun set<Config>(t: Config);
```

Reply:
I like the first API you suggest.

> The problem to both approaches is that we loose the ability to delegate the address in the default mode

Yeah, this is a tricky problem... But I think we can solve it with your first API + a small extension using the `Offer` module:

```
publicfun publish_new_config_with_delegate<Config>(t: Config, delegate: address)  {
   // ...
   Offer::create(modify_config_cap, delegate) // offer the modify_cap to delegate
}

public fun claim_delegated_modify_config(offer_address: address) {
  move_to_sender(Offer::redeem(offer_addres))
}
```

Reply:
I think we ended up very similar path! How do you feel about the current api?

Reply:
Getting there! I think that having both `DefaultModifyConfigCapability` and `ModifyConfigCapability` is a bit confusing.

I would prefer to have the simple model where `ModifyConfigCapability` as the *only*  way to modify an on-chain config if we can manage it.

Reply:
Just fixed it!